### DISCUSSION TITLE
[preset-env]: does it make sense to use it for browsers if they have started to roughen the browser version?

### DISCUSSION THREAD

**Discussion by budarin** [babel/babel#14975] [General]
Previously, we used `preset-env` for the last 2 versions of browsers.

Then, the server determined by the browser version from the `user-agent` whether this version is included in this group or not. 
If the browser was not part of this group, we redirected it to a page where we warned him that his browser had not been updated for a long time and offered other types of browsers to download

Now this scheme does not work and it makes no sense for `preset-env` to specify which versions we support in the `browserslist` since we cannot find out exactly the browser version.

It turns out that now we have to support the entire zoo of browsers and produce megabyte bundles :(
Or just ignore it and let the app crash on older versions of browsers!?

for example: real Chrome version - `105.0.5195.12`
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/`105.0.0.0` Safari/537.36

How now to build and deploy an application in such a situation?

**Comment by JLHwung** [comment]
> the server determined by the browser version from the user-agent whether this version is included in this group or not.

If the server is in your control. You can find a way to synchronize "this group" to `preset-env` tragets. If not, you may have to transpile all.

> a page where we warned him that his browser had not been updated for a long time and offered other types of browsers to download

You can make this page a dedicated app/website, transpiling its scripts down to ES3 or something.

**Reply by budarin** [reply]
> If the server is in your control. You can find a way to synchronize "this group" to `preset-env` tragets. If not, you may have to transpile all.

we did this until the last moment - we converted the expression "last 2 versions" into a list of specific browser versions during assembly, but now it doesn't work

> You can make this page a dedicated app/website, transpiling its scripts down to ES3 or something.

It will weigh a lot of megabytes - don't care about the performance?

**Reply by JLHwung** [reply]
> we converted the expression "last 2 versions" into a list of specific browser versions during assembly, but now it doesn't work

Your approach seems reasonable to me. It should work. Do you also update the `caniuse-lite`? Chances are the browser version is so new that the old `caniuse-lite` might not know that. If `caniuse-lite` is recent, please open an issue with a repo so we can take a look.

> It will weigh a lot of megabytes - don't care about the performance?

By a dedicated app/website, I mean the scripts of that page should only serve its own purpose: remind people to update their browsers. While the script could still weigh a lot of megabytes, that page could also be as light as a plain HTML. It's all up to you.

**Reply by budarin** [reply]
> Your approach seems reasonable to me. It should work. Do you also update the `caniuse-lite`? Chances are the browser version is so new that the old `caniuse-lite` might not know that. If `caniuse-lite` is recent, please open an issue with a repo so we can take a look.

thanks for mention it - we will do! )

**Reply by budarin** [reply]
> By a dedicated app/website, I mean the scripts of that page should only serve its own purpose: remind people to update their browsers. While the script could still weigh a lot of megabytes, that page could also be as light as a plain HTML. It's all up to you.

it's just about the size of js

**Reply by budarin** [reply]
> Your approach seems reasonable to me. It should work. Do you also update the `caniuse-lite`? Chances are the browser version is so new that the old `caniuse-lite` might not know that. If `caniuse-lite` is recent, please open an issue with a repo so we can take a look.

the approach does not work, not because of the relevance of the database, but because Chromium has stopped giving the exact version of the browser since the 100th version
real version - `105.0.5195.127`
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/`105.0.0.0` Safari/537.36
